thavalon-webapp/src/components/gameCreation.tsx

```diff
@@ -0,0 +1,325 @@
+    );
+}
+
+/**
+ * 
+ * @param showForm Sets whether or not a form should display
+ * @param setModalState Sets the modal state to the correct form that should display
+ */
+function CreateJoinButtons(showForm: Dispatch<SetStateAction<boolean>>, setModalState: Dispatch<SetStateAction<CreateJoinState>>): JSX.Element {
+    return (
+        <div className="create-join-buttons">
+            <Container>
+                <div className="center">
+                    <ButtonGroup vertical>
+                        <Button
+                            className="with-bottom-margin"
+                            variant="primary"
+                            onClick={() => {
+                                showForm(true);
+                                setModalState(CreateJoinState.JoinGame);
+                            }}>
+                            Join Game
+                        </Button>
+                        <Button
+                            variant="primary"
+                            onClick={() => {
+                                showForm(true);
+                                setModalState(CreateJoinState.CreateGame);
+                            }}>
+                            Create Game
+                        </Button>
+                    </ButtonGroup>
+                </div>
+            </Container>
+        </div >
+    );
+}
+
+/**
+ * Creates the "Create Game" form and handles submissions.
+ * @param props Props to set the form state and handle successes
+ */
+function CreateGameForm(props: FormProps): JSX.Element {
+    // Form to submit CreateGameData
+    const { register, handleSubmit } = useForm<CreateGameData>();
+    // State that tracks the form error message.
+    const [formErrorMsg, setFormErrorMsg] = useState("");
+    // State to check if we should redirect to a game.
+    const [redirectToGame, setRedirectToGame] = useState(false);
+    // State to maintain the friend code to send to the redirect link.
+    const [friendCode, setFriendCode] = useState("");
+    // State to maintain the socketUrl to send to the lobby component.
+    const [socketUrl, setSocketUrl] = useState("");
+
+    /**
+     * Handles a submission to create a game by the player. Will create the game
+     * and then auto-join the game, if there are no errors.
+     * @param data The CreateGameData from the submitting form
+     */
+    async function onCreateGameSubmit(data: CreateGameData) {
+        const createGameResponse = await connection.createGame();
+        if (createGameResponse.result === false) {
+            setFormErrorMsg(createGameResponse.message);
+            return;
+        }
+
+        const friendCode = createGameResponse.message;
+        setFriendCode(friendCode);
+        const joinGameResponse = await connection.joinGame(friendCode, data.displayName);
+        if (joinGameResponse.result === false) {
+            setFormErrorMsg(joinGameResponse.message);
+            return;
+        }
```

Comment:
We could have this skip creating a game if the `friendCode` state is non-empty. That way, if the game is created successfully but joining fails, the player can try joining again